KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift

```diff
@@ -412,18 +413,11 @@ final class KarhooCheckoutPresenter: CheckoutPresenter {
     
     private func getPaymentNonceAccordingToAuthState() -> String? {
         switch Karhoo.configuration.authenticationMethod() {
-        case .tokenExchange(settings: _), .karhooUser: return retrievePaymentNonce()
+        case .tokenExchange(settings: _), .karhooUser: return view?.getPaymentNonce()?.nonce
         default: return view?.getPaymentNonce()?.nonce
         }
     }
-    
-    private func retrievePaymentNonce() -> String? {
-        if userService.getCurrentUser()?.paymentProvider?.provider.type == .braintree {
-            return userService.getCurrentUser()?.nonce?.nonce
```

Comment:
It looks like we lost this piece of logic for Braintree

Reply:
It is not needed. In both cases the nonce can be retrieved using `userService.getCurrentUser()?.nonce?.nonce`. Therefore, `if` statement is not needed.

Reply:
Yes, but in the method above (where `retrievePaymentNonce()` used to be called) the nonce is retrieved from the view, not the `userService`

Reply:
You are right. I've re-added `retrivePaymentNonce()` method and changed all calls to nonce in the class to use this function.